NullPointerException when using kv() in input defaults with Flow triggers

### Describe the issue


## Issue Summary
When a flow with a Flow-type trigger uses the `kv()` function in an input's `defaults` field, it causes a `NullPointerException` during trigger evaluation because the flow context is not available yet.

## Error Stack Trace

```
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "java.util.Map.get(Object)" because "flow" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	...
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.Map.get(Object)" because "flow" is null
	at io.kestra.core.runners.pebble.functions.KvFunction.execute(KvFunction.java:44)
	at io.pebbletemplates.pebble.node.expression.FunctionOrMacroInvocationExpression.applyFunction(FunctionOrMacroInvocationExpression.java:46)
	...
	at io.kestra.core.runners.FlowInputOutput.resolveDefaultValue(FlowInputOutput.java:362)
	at io.kestra.core.runners.RunVariables$DefaultBuilder.lambda$build$4(RunVariables.java:290)
	...
	at io.kestra.executor.FlowTriggerService.lambda$computeExecutionsFromFlowTriggers$11(FlowTriggerService.java:72)
```

## Root Cause

The execution sequence is:
1. Parent flow completes
2. `FlowTriggerService.computeExecutionsFromFlowTriggers()` evaluates which flows to trigger
3. To evaluate triggers, it creates a `RunContext` via `RunContextFactory.of()`
4. This requires resolving input default values via `FlowInputOutput.resolveDefaultValue()`
5. **At this point, no execution has been created yet, so `flow` context is null**
6. The `kv()` function tries to access `flow.get("namespace")` and `flow.get("id")` → NPE

## Minimal Reproducible Example

### Parent Flow (Webhook Trigger)

```yaml
id: parent_flow
namespace: prod

tasks:
  - id: log
    type: io.kestra.plugin.core.log.Log
    message: "Parent flow executed"

triggers:
  - id: webhook
    type: io.kestra.plugin.core.trigger.Webhook
    key: test123
```

### Child Flow (Flow Trigger - CAUSES ERROR)

```yaml
id: child_flow
namespace: prod

inputs:
  - id: llm_model
    type: STRING
    required: true
    defaults: "{{ kv('my-model-key') }}"  # ← This causes NullPointerException

tasks:
  - id: log
    type: io.kestra.plugin.core.log.Log
    message: "Using model: {{ inputs.llm_model }}"

triggers:
  - id: parent_flow_trigger
    type: io.kestra.plugin.core.trigger.Flow
    inputs:
      data: "{{ trigger.outputs }}"
    conditions:
      - type: io.kestra.plugin.core.condition.ExecutionFlow
        namespace: prod
        flowId: parent_flow
```

### Setup

1. Create a KV store entry: key `my-model-key`, value `gpt-4`
2. Create both flows
3. Trigger the parent flow via webhook
4. Observe the error when `FlowTriggerService` tries to trigger the child flow

### Environment

- Kestra Version: 1.0.7

--- a/core/src/main/java/io/kestra/core/runners/RunVariables.java
+++ b/core/src/main/java/io/kestra/core/runners/RunVariables.java
@@ -10,7 +10,6 @@ import io.kestra.core.models.flows.FlowInterface;
 import io.kestra.core.models.flows.Input;
 import io.kestra.core.models.flows.State;
 import io.kestra.core.models.flows.input.SecretInput;
-import io.kestra.core.models.property.Property;
 import io.kestra.core.models.property.PropertyContext;
 import io.kestra.core.models.tasks.Task;
 import io.kestra.core.models.triggers.AbstractTrigger;
@@ -100,7 +99,7 @@ public final class RunVariables {
      * @return a new immutable {@link Map}.
      */
     static Map<String, Object> of(final AbstractTrigger trigger) {
-        return ImmutableMap.of(
+        return Map.of(
             "id", trigger.getId(),
             "type", trigger.getType()
         );
@@ -282,12 +281,15 @@ public final class RunVariables {
                 }
 
                 if (flow != null && flow.getInputs() != null) {
+                    // Create a new PropertyContext with 'flow' variables which are required by some pebble expressions.
+                    PropertyContextWithVariables context = new PropertyContextWithVariables(propertyContext, Map.of("flow", RunVariables.of(flow)));
+                    
                     // we add default inputs value from the flow if not already set, this will be useful for triggers
                     flow.getInputs().stream()
                         .filter(input -> input.getDefaults() != null && !inputs.containsKey(input.getId()))
                         .forEach(input -> {
                             try {
-                                inputs.put(input.getId(), FlowInputOutput.resolveDefaultValue(input, propertyContext));
+                                inputs.put(input.getId(), FlowInputOutput.resolveDefaultValue(input, context));
                             } catch (IllegalVariableEvaluationException e) {
                                 // Silent catch, if an input depends on another input, or a variable that is populated at runtime / input filling time, we can't resolve it here.
                             }
@@ -391,4 +393,20 @@ public final class RunVariables {
     }
 
     private RunVariables(){}
+    
+    private record PropertyContextWithVariables(
+        PropertyContext delegate,
+        Map<String, Object> variables
+    ) implements PropertyContext {
+        
+        @Override
+        public String render(String inline, Map<String, Object> variables) throws IllegalVariableEvaluationException {
+            return delegate.render(inline, variables.isEmpty() ? this.variables : variables);
+        }
+        
+        @Override
+        public Map<String, Object> render(Map<String, Object> inline, Map<String, Object> variables) throws IllegalVariableEvaluationException {
+            return delegate.render(inline, variables.isEmpty() ? this.variables : variables);
+        }
+    }
 }

--- a/core/src/test/java/io/kestra/core/runners/FlowInputOutputTest.java
+++ b/core/src/test/java/io/kestra/core/runners/FlowInputOutputTest.java
@@ -13,13 +13,18 @@ import io.kestra.core.models.flows.input.StringInput;
 import io.kestra.core.models.property.Property;
 import io.kestra.core.secret.SecretNotFoundException;
 import io.kestra.core.secret.SecretService;
+import io.kestra.core.services.KVStoreService;
 import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.storages.kv.InternalKVStore;
+import io.kestra.core.storages.kv.KVStore;
+import io.kestra.core.storages.kv.KVValue;
 import io.kestra.core.utils.IdUtils;
 import io.micronaut.http.MediaType;
 import io.micronaut.http.multipart.CompletedFileUpload;
 import io.micronaut.http.multipart.CompletedPart;
 import io.micronaut.test.annotation.MockBean;
 import jakarta.inject.Inject;
+import org.jetbrains.annotations.Nullable;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.reactivestreams.Publisher;
@@ -41,6 +46,7 @@ import static org.assertj.core.api.Assertions.assertThat;
 class FlowInputOutputTest {
     
     private static final String TEST_SECRET_VALUE = "test-secret-value";
+    private static final String TEST_KV_VALUE = "test-kv-value";
     
     static final Execution DEFAULT_TEST_EXECUTION = Execution.builder()
         .id(IdUtils.create())
@@ -61,6 +67,21 @@ class FlowInputOutputTest {
             @Override
             public String findSecret(String tenantId, String namespace, String key) throws SecretNotFoundException {
                 return TEST_SECRET_VALUE;
+            }
+        };
+    }
+    
+    @MockBean(KVStoreService.class)
+    KVStoreService testKVStoreService() {
+        return new KVStoreService() {
+            @Override
+            public KVStore get(String tenant, String namespace, @Nullable String fromNamespace) {
+                return new InternalKVStore(tenant, namespace, storageInterface) {
+                    @Override
+                    public Optional<KVValue> getValue(String key) {
+                        return Optional.of(new KVValue(TEST_KV_VALUE));
+                    }
+                };
             }
         };
     }
@@ -352,6 +373,23 @@ class FlowInputOutputTest {
         
         // Then
         Assertions.assertEquals(TEST_SECRET_VALUE, results.get("input"));
+    }
+    
+    @Test
+    void shouldEvaluateExpressionOnDefaultsUsingKVFunction() {
+        // Given
+        StringInput input = StringInput.builder()
+            .id("input")
+            .type(Type.STRING)
+            .defaults(Property.ofExpression("{{ kv('???') }}"))
+            .required(false)
+            .build();
+        
+        // When
+        Map<String, Object> results = flowInputOutput.readExecutionInputs(List.of(input), null, DEFAULT_TEST_EXECUTION, Mono.empty()).block();
+        
+        // Then
+        assertThat(results.get("input")).isEqualTo(TEST_KV_VALUE);
     }
     
     @Test

--- a/core/src/test/java/io/kestra/core/runners/RunVariablesTest.java
+++ b/core/src/test/java/io/kestra/core/runners/RunVariablesTest.java
@@ -1,9 +1,11 @@
 package io.kestra.core.runners;
 
-import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.junit.annotations.KestraTest;
 import io.kestra.core.models.executions.Execution;
 import io.kestra.core.models.flows.DependsOn;
 import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.flows.FlowInterface;
+import io.kestra.core.models.flows.GenericFlow;
 import io.kestra.core.models.flows.Type;
 import io.kestra.core.models.flows.input.BoolInput;
 import io.kestra.core.models.property.Property;
@@ -11,25 +13,55 @@ import io.kestra.core.models.property.PropertyContext;
 import io.kestra.core.models.tasks.Task;
 import io.kestra.core.models.triggers.AbstractTrigger;
 import io.kestra.core.runners.pebble.PebbleEngineFactory;
+import io.kestra.core.services.KVStoreService;
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.storages.kv.InternalKVStore;
+import io.kestra.core.storages.kv.KVStore;
+import io.kestra.core.storages.kv.KVValue;
+import io.kestra.core.tenant.TenantService;
 import io.kestra.core.utils.IdUtils;
 import io.micronaut.context.ApplicationContext;
+import io.micronaut.test.annotation.MockBean;
+import jakarta.inject.Inject;
+import org.jetbrains.annotations.Nullable;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.mockito.Mockito;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+@KestraTest
 class RunVariablesTest {
     
-    private final PropertyContext propertyContext = Mockito.mock(PropertyContext.class);
+    @Inject
+    VariableRenderer renderer;
+    
+    @Inject
+    StorageInterface storageInterface;
+    
+    @MockBean(KVStoreService.class)
+    KVStoreService testKVStoreService() {
+        return new KVStoreService() {
+            @Override
+            public KVStore get(String tenant, String namespace, @Nullable String fromNamespace) {
+                return new InternalKVStore(tenant, namespace, storageInterface) {
+                    @Override
+                    public Optional<KVValue> getValue(String key) {
+                        return Optional.of(new KVValue("value"));
+                    }
+                };
+            }
+        };
+    }
     
     @Test
     @SuppressWarnings("unchecked")
     void shouldGetEmptyVariables() {
-        Map<String, Object> variables = new RunVariables.DefaultBuilder().build(new RunContextLogger(), propertyContext);
+        Map<String, Object> variables = new RunVariables.DefaultBuilder().build(new RunContextLogger(), PropertyContext.create(renderer));
         assertThat(variables.size()).isEqualTo(3);
         assertThat((Map<String, Object>) variables.get("envs")).isEqualTo(Map.of());
         assertThat((Map<String, Object>) variables.get("globals")).isEqualTo(Map.of());
@@ -46,7 +78,7 @@ class RunVariablesTest {
                 .revision(42)
                 .build()
             )
-            .build(new RunContextLogger(), propertyContext);
+            .build(new RunContextLogger(), PropertyContext.create(renderer));
         Assertions.assertEquals(Map.of(
             "id", "id-value",
             "namespace", "namespace-value",
@@ -65,7 +97,7 @@ class RunVariablesTest {
                 .tenantId("tenant-value")
                 .build()
             )
-            .build(new RunContextLogger(), propertyContext);
+            .build(new RunContextLogger(), PropertyContext.create(renderer));
         Assertions.assertEquals(Map.of(
             "id", "id-value",
             "namespace", "namespace-value",
@@ -88,7 +120,7 @@ class RunVariablesTest {
                     return "type-value";
                 }
             })
-            .build(new RunContextLogger(), propertyContext);
+            .build(new RunContextLogger(), PropertyContext.create(renderer));
         Assertions.assertEquals(Map.of("id", "id-value", "type", "type-value"), variables.get("task"));
     }
 
@@ -106,7 +138,7 @@ class RunVariablesTest {
                     return "type-value";
                 }
             })
-            .build(new RunContextLogger(), propertyContext);
+            .build(new RunContextLogger(), PropertyContext.create(renderer));
         Assertions.assertEquals(Map.of("id", "id-value", "type", "type-value"), variables.get("trigger"));
     }
 
@@ -115,7 +147,7 @@ class RunVariablesTest {
     void shouldGetKestraConfiguration() {
         Map<String, Object> variables = new RunVariables.DefaultBuilder()
             .withKestraConfiguration(new RunVariables.KestraConfiguration("test", "http://localhost:8080"))
-            .build(new RunContextLogger(), propertyContext);
+            .build(new RunContextLogger(), PropertyContext.create(renderer));
         assertThat(variables.size()).isEqualTo(4);
         Map<String, Object> kestra = (Map<String, Object>) variables.get("kestra");
         assertThat(kestra).hasSize(2);
@@ -124,7 +156,7 @@ class RunVariablesTest {
     }
 
     @Test
-    void nonResolvableDynamicInputsShouldBeSkipped() throws IllegalVariableEvaluationException {
+    void nonResolvableDynamicInputsShouldBeSkipped() {
         VariableRenderer.VariableConfiguration mkVariableConfiguration = Mockito.mock(VariableRenderer.VariableConfiguration.class);
         ApplicationContext mkApplicationContext = Mockito.mock(ApplicationContext.class);
         Map<String, Object> variables = new RunVariables.DefaultBuilder()
@@ -144,5 +176,24 @@ class RunVariablesTest {
         Assertions.assertEquals(Map.of(
             "a", true
         ), variables.get("inputs"));
+    }
+    
+    @Test
+    void shouldBuildVariablesGivenFlowWithInputHavingDefaultPebbleExpression() {
+        FlowInterface flow = GenericFlow.fromYaml(TenantService.MAIN_TENANT, """
+            id: id-value
+            namespace: namespace-value
+            inputs:
+            - id: input
+              type: STRING
+              defaults: "{{ kv('???') }}"
+            """);
+        
+        Map<String, Object> variables = new RunVariables.DefaultBuilder()
+            .withFlow(flow)
+            .withExecution(Execution.builder().id(IdUtils.create()).build())
+            .build(new RunContextLogger(), PropertyContext.create(renderer));
+        
+        assertThat(variables.get("inputs")).isEqualTo(Map.of("input", "value"));
     }
 }

